opencog/openpsi/utilities.scm

```diff
@@ -326,13 +326,13 @@
 		result
 		; else check if entity is an evaluation or predicate or schema
 		(let ((type (cog-type entity)))
-			(if (or (equal? type 'PredicateNode)
-				   (equal? type 'GroundedPredicateNode)
+			(if (or (equal? type 'GroundedPredicateNode)
 				   (equal? type 'DefinedPredicateNode))
 				(set! result (cog-evaluate! (Evaluation entity (List)))))
-			(if (or (equal? type 'SchemaNode)
-            		(equal? type 'GroundedSchemaNode)
-            	    (equal? type 'DefinedSchemaeNode))
+			(if (equal? type 'PredicateNode)
+				(set! result (cog-tv entity)))
```

Comment:
On Mon, Aug 1, 2016 at 11:04 AM, Eddie Monroe <notifications@github.com>
wrote:

> For a Predicate with no arguments, I guess the node TV might represent
> something like it's average value over time, while the EvaluationLink tv
> would be its current value.


This suggestion seems problematic...

One could have two predicates with different time-references e.g.

My_happiness_now

and

My_happiness_generally

and in each case, if the predicates have no arguments, you could pretty
much assign them however you want .. I don't see the problem with that...

Using the difference between the  wrapping EvaluationLink TV and the node
TV to indicate different scopes of time seems problematic, i.e. one is
hiding semantics that should be made explicit...


-- 
Ben Goertzel, PhD
http://goertzel.org

Super-benevolent super-intelligence is the thought the Global Brain is
currently struggling to form...